frontend-react/src/components/MessageTracker/MessageReceivers.tsx

```diff
@@ -1,62 +1,511 @@
-                    ></DetailItem>
-                    <DetailItem
-                        item={"Incoming File Name"}
-                        content={receiver.fileName}
-                    ></DetailItem>
-                    <DetailItem
-                        item={"Incoming File URL"}
-                        content={receiver.fileUrl}
-                    ></DetailItem>
-                    <DetailItem
-                        item={"Date/Time Submitted"}
-                        content={formattedDateFromTimestamp(
-                            receiver.createdAt,
-                            "MMMM DD YYYY"
+                    <thead>
+                        <tr>
+                            <MessageReceiversCol
+                                columnHeaderTitle={ColumnDataEnum.Name}
+                                activeColumn={activeColumn}
+                                setActiveColumn={setActiveColumn}
+                                activeColumnSortOrder={activeColumnSortOrder}
+                                setActiveColumnSortOrder={
+                                    setActiveColumnSortOrder
+                                }
+                                rowSpan={2}
+                            />
+                            <MessageReceiversCol
+                                columnHeaderTitle={ColumnDataEnum.Service}
+                                activeColumn={activeColumn}
+                                setActiveColumn={setActiveColumn}
+                                activeColumnSortOrder={activeColumnSortOrder}
+                                setActiveColumnSortOrder={
+                                    setActiveColumnSortOrder
+                                }
+                                rowSpan={2}
+                            />
+                            <MessageReceiversCol
+                                columnHeaderTitle={ColumnDataEnum.Date}
+                                activeColumn={activeColumn}
+                                setActiveColumn={setActiveColumn}
+                                activeColumnSortOrder={activeColumnSortOrder}
+                                setActiveColumnSortOrder={
+                                    setActiveColumnSortOrder
+                                }
+                                rowSpan={2}
+                            />
+                            <MessageReceiversCol
+                                columnHeaderTitle={ColumnDataEnum.ReportId}
+                                activeColumn={activeColumn}
+                                setActiveColumn={setActiveColumn}
+                                activeColumnSortOrder={activeColumnSortOrder}
+                                setActiveColumnSortOrder={
+                                    setActiveColumnSortOrder
+                                }
+                                rowSpan={2}
+                            />
+                            <th colSpan={3}>
```

Comment:
Definitely not for now, but along the lines of what we were talking about for coming up with a standard structure to pass to the table, I wonder if we could do something like recursively nest them so we could programmatically iterate over them when rendering:

```tsx
const headers = [{
    title: "Name"
}, {
    title: "File Location",
    headers: [{
        title: "Transport Results"
    }, {
        title: "Main"
    }, {
        title: "Sub"
    }]
}];
```

idk something like that 🤷

Reply:
Agree and could have other properties for each object i.e.
{
        field: 'reportId',
        label: 'Report Id',
        colSpan: ColSpanValue,
        sortable: true,
        sortField: sortFields.name
      }